json-schema-generator/src/main/java/io/micronaut/jsonschema/generator/SourceGenerator.java

```diff
@@ -0,0 +1,638 @@
+        String finalSchemaName = getClassName(schemaName);
+
+        // save all definition and oneOf types
+        if (jsonSchema.hasOneOf()) {
+            jsonSchema.getOneOf().forEach(oneOf -> {
+                if (oneOf.has$ref()) {
+                    String ref = oneOf.get$ref();
+                    if (ref.indexOf("#") == 0) {
+                        ref = inputFileName + ref;
+                    }
+                    context.addOneOf(ref);
+                } else {
+                    context.addOneOf(oneOf);
+                }
+            });
+        }
+        if (jsonSchema.has$defs()) {
+            jsonSchema.get$defs().forEach((key, value) -> {
+                if (key.equals("//")) {
+                    if (!jsonSchema.hasDescription()) {
+                        jsonSchema.setDescription(String.valueOf(value));
+                    } else {
+                        jsonSchema.setDescription(jsonSchema.getDescription() + "<br>" + value);
+                    }
+                } else if (value.hasOneOf() && jsonSchema.hasDiscriminator()) {
+                    // WARNING: assumes the same interface as top level schema
+                    context.addDefinition(inputFileName + DEF_SCHEMA_REF_PREFIX + key, TypeDef.THIS, true);
+                } else {
+                    context.addDefinition(inputFileName + DEF_SCHEMA_REF_PREFIX + key, value);
+                }
+            });
+        }
+        context.addDefinition(inputFileName + "#/" + finalSchemaName, jsonSchema);
+    }
+
+    private File generateDefinitions(Schema jsonSchema, Path outputPath, String packageName) throws IOException {
+        // generate top level schema
+        String schemaName = jsonSchema.hasTitle() ? jsonSchema.getTitle() : inputFileName.substring(0, inputFileName.indexOf('.'));
+        schemaName = getClassName(schemaName);
+
+        File topLevelObject = generateFromSchema(jsonSchema, outputPath, packageName, schemaName);
+
+        // generate classes in definitions and oneOfs
+        if (jsonSchema.has$defs()) {
+            jsonSchema.get$defs().entrySet()
+                .stream()
+                .filter(definition -> !definition.getKey().equals("//") && context.isDefinitionClass(inputFileName + DEF_SCHEMA_REF_PREFIX + definition.getKey()))
+                .forEach(definition -> {
+                    try {
+                        var className = getClassName(definition.getKey());
+                        generateFromSchema(definition.getValue(), outputPath, packageName, className);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+        }
+        for (Map.Entry<String, Schema> oneOf : context.getOneOfsToGenerate()) {
+            String className = oneOf.getKey().substring(oneOf.getKey().lastIndexOf('/') + 1);
```

Comment:
Should this use the `getClassName()` method?

Reply:
Not really, the key was generated already using the `getClassName()` method so I just need to extract it from the complete address.